phpseclib/System/SSH/Agent/Identity.php

```diff
@@ -115,29 +114,17 @@ function getPublicKey($format = null)
     }
 
     /**
-     * Set Signature Mode
-     *
-     * Doesn't do anything as ssh-agent doesn't let you pick and choose the signature mode. ie.
-     * ssh-agent's only supported mode is \phpseclib\Crypt\RSA::SIGNATURE_PKCS1
-     *
-     * @param int $mode
-     * @access public
-     */
-    function setSignatureMode($mode)
-    {
-    }
-
-    /**
      * Create a signature
      *
      * See "2.6.2 Protocol 2 private key signature request"
      *
      * @param string $message
+     * @param int|bool $padding
      * @return string
      * @throws \RuntimeException on connection errors
      * @access public
      */
-    function sign($message)
+    function sign($message, $padding = false)
```

Comment:
Should this be `function sign($message, $padding = RSA::PADDING_NONE)`, to have the same type as with RSA::sign()? (Would require `use phpseclib\Crypt\RSA;` at the top of the file.)